### PR TITLE
Make parameters importable from a directory having "pybamm" in its name

### LINKED ISSUE #1918: Having "pybamm" in a directory's name throws import error
For the import error, the issue is with your folder's name (as it contains the string `pybamm` in it). The code tries to treat it as the main `pybamm` folder -
https://github.com/pybamm-team/PyBaMM/blob/33c6d787d1455ac56a08ab228d2c51b9ba65240c/pybamm/util.py#L290-L292

_Originally posted by @Saransh-cpp in https://github.com/pybamm-team/PyBaMM/discussions/1639#discussioncomment-2042134_

### PR DESCRIPTION
**Pull request by Saransh-cpp**
# Description

Check for `pybamm/input/parameters` instead of `pybamm` while trying to import parameters from the base package.

Fixes #1918 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [X] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [X] No style issues: `$ flake8`
- [X] All tests pass: `$ python run-tests.py --unit`
- [X] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Comment by priyanshuone6** [comment]
Could you add some tests

**Comment by Saransh-cpp** [comment]
> Could you add some tests

For some reason, the tests for the modified line don't work. I tried doing -
```py 
        pybamm.set_logging_level("DEBUG")
    
        import subprocess

        # create a new lithium_ion folder in the root PyBaMM directory
        subprocess.run(["pybamm_edit_parameter", "lithium_ion"])
       
        # path for a function in the created directory -> x/y/z/PyBaMM/lithium_ion/negative_electrode/ ....
        test_path = os.path.join(
            os.getcwd(),
            "lithium_ion",
            "negative_electrodes",
            "graphite_Chen2020",
            "graphite_LGM50_electrolyte_exchange_current_density_Chen2020.py",
        )

        # check if the lithium_ion directory is present in PyBaMM -> It is
        print(os.listdir(os.getcwd()))

        # load the function
        func = pybamm.load_function(test_path)
        self.assertEqual(
            func,
lithium_ion.negative_electrodes.graphite_Chen2020.graphite_LGM50_electrolyte_exchange_current_density_Chen2020.graphite_LGM50_electrolyte_exchange_current_density_Chen2020,  # noqa
      )
```

This shows me -
```
2022-01-29 17:36:39,496 - [DEBUG] util.load_function(312): Importing function 'graphite_LGM50_electrolyte_exchange_current_density_Chen2020' from file 'C:\Users\Saransh\Saransh_softwares\OpenSource\Python\PyBaMM\lithium_ion\negative_electrodes\graphite_Chen2020\graphite_LGM50_electrolyte_exchange_current_density_Chen2020' via path 'lithium_ion.negative_electrodes.graphite_Chen2020.graphite_LGM50_electrolyte_exchange_current_density_Chen2020'

ModuleNotFoundError: No module named 'lithium_ion'
```

Importing parameters from anywhere works completely fine outside of the testing suite.

Apparently, importing parameters from anywhere else than `pybamm/input/parameters/` is not tested in `PyBaMM`. I couldn't really figure out why is it not working in tests but is working when used elsewhere.

File: pybamm/util.py [hunk 1]
```diff
@@ -288,25 +288,25 @@ def load_function(filename):
...
         dir_path = os.path.split(filename)[0]
         os.chdir(dir_path)
         root_path = filename.replace(os.getcwd(), "")
-        root_path = root_path[1:]
     else:
         root_path = filename
 
+    # getcwd() returns "C:\\" when in the root drive and "C:\\a\\b\\c" otherwise
+    if root_path[0] == "\\" or root_path[0] == "/":
+        root_path = root_path[1:]
```
Comment (tinosulzer) [inline pybamm/util.py:308]: is `"\\"` a single character?

Comment (Saransh-cpp) [reply pybamm/util.py:308]: Yes

**Comment by tinosulzer** [comment]
> Apparently, importing parameters from anywhere else than pybamm/input/parameters/ is not tested in PyBaMM. I couldn't really figure out why is it not working in tests but is working when used elsewhere.

Yeah this is strange, I get this error locally as well

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,7 +13,7 @@
 
 ## Bug fixes
 
--   Parameters can now be imported from any given path in `Windows` ([#1900](https://github.com/pybamm-team/PyBaMM/pull/1900))
+-   Parameters can now be imported from any given path ([#1900](https://github.com/pybamm-team/PyBaMM/pull/1900), [#1919](https://github.com/pybamm-team/PyBaMM/pull/1919))
```
Comment (priyanshuone6) [inline CHANGELOG.md:16]: Could you merge develop and move this to the unreleased section

**Comment by Saransh-cpp** [comment]
> Did you want to add a test for this? Or shall I merge?

I found the issue with the failing tests -
- The tests pass when run using `python run-tests.py --unit` as the `.../..../PyBaMM` directory is in `sys.path`
- The tests fail when `test_utils.py` is run individually, as now `..../..../PyBaMM/tests/unit` is in `sys.path` and not `.../..../PyBaMM`

**Review by tinosulzer — APPROVED** [review]
Thanks!

**Pull request merged**